packages/legacy/core/App/screens/Connection.tsx

```diff
@@ -72,102 +69,98 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
+  //     }, connTimerDelay)
+
+  //     dispatch({ isInitialized: true })
+  //   }
+  // }
+
+  // const abortTimer = () => {
+  //   if (timerRef.current) {
+  //     clearTimeout(timerRef.current)
+  //     timerRef.current = null
+  //   }
+  // }
 
-  const setModalVisible = (value: boolean) => {
-    setState((prev) => ({ ...prev, isVisible: value }))
-  }
-  const setIsInitialized = (value: boolean) => {
-    setState((prev) => ({ ...prev, isInitialized: value }))
-  }
-  const setShouldShowDelayMessage = (value: boolean) => {
-    setState((prev) => ({ ...prev, shouldShowDelayMessage: value }))
-  }
-  useEffect(() => {
-    if (autoRedirectConnectionToHome && shouldShowDelayMessage && connectionIsActive && !notificationRecord) {
-      setShouldShowDelayMessage(false)
-      setModalVisible(false)
-      navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
-    }
-  }, [shouldShowDelayMessage])
-  const abortTimer = () => {
-    if (timerRef.current) {
-      clearTimeout(timerRef.current)
-      timerRef.current = null
-    }
-  }
   const onDismissModalTouched = () => {
-    setShouldShowDelayMessage(false)
-    setModalVisible(false)
+    dispatch({ shouldShowDelayMessage: false, isVisible: false })
     navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
   }
-  const startTimer = () => {
-    if (!isInitialized) {
-      timerRef.current = setTimeout(() => {
-        setShouldShowDelayMessage(true)
-        timerRef.current = null
-      }, connTimerDelay)
-      setIsInitialized(true)
-    }
-  }
 
   useEffect(() => {
-    if (shouldShowDelayMessage && !notificationRecord) {
-      AccessibilityInfo.announceForAccessibility(t('Connection.TakingTooLong'))
+    // Wait for OOB proof notification to arrive.
+    if (!connection) {
+      return
     }
-  }, [shouldShowDelayMessage])
 
-  useFocusEffect(
-    useCallback(() => {
-      startTimer()
-      return () => abortTimer
-    }, [])
-  )
+    // Note: V1 connections are not required to enter the `Completed` state and may
+    // remain at `RequestSent` indefinitely. This is a known issue that is addressed
+    // in higher protocol versions.
+    if (connection.state === DidExchangeState.Completed || connection.state === DidExchangeState.RequestSent) {
+      dispatch({ connectionIsActive: true })
+      agent?.mediationRecipient.initiateMessagePickup()
 
-  useEffect(() => {
-    if (notificationRecord) {
-      switch (notificationRecord.type) {
-        case 'CredentialRecord':
-          navigation.navigate(Screens.CredentialOffer, { credentialId: notificationRecord.id })
-          break
-        case 'ProofRecord':
-          navigation.navigate(Screens.ProofRequest, { proofId: notificationRecord.id })
-          break
-        default:
-          throw new Error('Unhandled notification type')
-      }
+      // No goal code, we don't know what to expect next,
+      // navigate to the contact details.
```

Comment:
It really looks like it still navigates to the contact details screen

Reply:
Yes, didn't push my change + I just updated the docs. Should be fixed now.